tests/ops/qutrit/test_qutrit_matrix_ops.py

```diff
@@ -0,0 +1,376 @@
+
+U_thadamard_01 = np.multiply(
+    1 / np.sqrt(2),
+    np.array(
+        [[1, 1, 0], [1, -1, 0], [0, 0, np.sqrt(2)]],
+    ),
+)
+
+U_tswap = np.array(
+    [
+        [1, 0, 0, 0, 0, 0, 0, 0, 0],
+        [0, 0, 0, 1, 0, 0, 0, 0, 0],
+        [0, 0, 0, 0, 0, 0, 1, 0, 0],
+        [0, 1, 0, 0, 0, 0, 0, 0, 0],
+        [0, 0, 0, 0, 1, 0, 0, 0, 0],
+        [0, 0, 0, 0, 0, 0, 0, 1, 0],
+        [0, 0, 1, 0, 0, 0, 0, 0, 0],
+        [0, 0, 0, 0, 0, 1, 0, 0, 0],
+        [0, 0, 0, 0, 0, 0, 0, 0, 1],
+    ],
+    dtype=np.complex128,
+)
+
+
+class TestQutritUnitary:
+    """Tests for the QutritUnitary class."""
+
+    def test_qutrit_unitary_noninteger_pow(self):
+        """Test QutritUnitary raised to a non-integer power raises an error."""
+        op = qml.QutritUnitary(U_thadamard_01, wires="a")
+
+        with pytest.raises(qml.operation.PowUndefinedError):
+            op.pow(0.123)
+
+    # TODO: Check with Olivia about broadcasted QutritUnitary
+    # def test_qutrit_unitary_noninteger_pow_broadcasted(self):
+    #     """Test broadcasted QutritUnitary raised to a non-integer power raises an error."""
+    #     U = np.array(
+    #         [
+    #             U_thadamard_01,
+    #             U_thadamard_01,
+    #         ]
+    #     )
+
+    #     op = qml.QutritUnitary(U, wires="a")
+
+    #     with pytest.raises(qml.operation.PowUndefinedError):
+    #         op.pow(0.123)
+
+    @pytest.mark.parametrize("n", (1, 3, -1, -3))
+    def test_qutrit_unitary_pow(self, n):
+        """Test qutrit unitary raised to an integer power."""
+        op = qml.QutritUnitary(U_thadamard_01, wires="a")
+        new_ops = op.pow(n)
+
+        assert len(new_ops) == 1
+        assert new_ops[0].wires == op.wires
+
+        mat_to_pow = qml.math.linalg.matrix_power(qml.matrix(op), n)
+        new_mat = qml.matrix(new_ops[0])
+
+        assert qml.math.allclose(mat_to_pow, new_mat)
+
+    # TODO: Check with Olivia about broadcasted QutritUnitary
+    # @pytest.mark.parametrize("n", (1, 3, -1, -3))
+    # def test_qutrit_unitary_pow_broadcasted(self, n):
+    #     """Test broadcasted qutrit unitary raised to an integer power."""
+    #     U = np.array(
+    #         [
+    #             U_thadamard_01,
+    #             U_thadamard_01,
+    #         ]
+    #     )
+
+    #     op = qml.QutritUnitary(U, wires="a")
+    #     new_ops = op.pow(n)
+
+    #     assert len(new_ops) == 1
+    #     assert new_ops[0].wires == op.wires
+
+    #     mat_to_pow = qml.math.linalg.matrix_power(qml.matrix(op), n)
+    #     new_mat = qml.matrix(new_ops[0])
+
+    #     assert qml.math.allclose(mat_to_pow, new_mat)
+
+    interface_and_decomp_data = [
+        (U_thadamard_01, 1),
+        (U_tswap, 2),
+        (unitary_group.rvs(3, random_state=10), 1),
+        (unitary_group.rvs(9, random_state=10), 2),
+        (np.eye(3), 1),
+        (np.eye(9), 2),
+        # (np.tensordot([1j, -1, 1], U_thadamard_01, axes=0), 1)
```

Comment:
This test uses a batch of unitaries rather than a single unitary matrix, and since that has not yet been implemented for `QutritUnitary`, it is not required. I will go back and remove that test for now from the PR and add it eventually once `QutritUnitary` supports batches of matrices.